From 9a6b4558bb7f470152ab00692010f07fd0bfa8f8 Mon Sep 17 00:00:00 2001
From: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date: Thu, 17 Dec 2020 01:19:33 +0100
Subject: [PATCH] mali: support building against 5.10

Starting from Linux version 5.10.0 <linux/dma-contiguous.h> has been
removed and its content moved to <linux/dma-mapping.h> and this leads
to build failure with Linux 5.10. So let's include
<include/dma-contiguous.h> only if we're building with Linux
version < 5.10.0.

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
---
 src/devicedrv/mali/platform/sunxi/sunxi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/devicedrv/mali/platform/sunxi/sunxi.c b/src/devicedrv/mali/platform/sunxi/sunxi.c
index 74ebe0b..a8df0d9 100644
--- a/src/devicedrv/mali/platform/sunxi/sunxi.c
+++ b/src/devicedrv/mali/platform/sunxi/sunxi.c
@@ -3,7 +3,10 @@
 #include <linux/clkdev.h>
 #include <linux/cma.h>
 #include <linux/delay.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
 #include <linux/dma-contiguous.h>
+#endif
 #include <linux/dma-mapping.h>
 #include <linux/io.h>
 #include <linux/kernel.h>
-- 
2.25.1

